okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -151,6 +130,24 @@ private long remainingNanos(long now) {
   protected void timedOut() {
   }
 
+  @Override
+  public final long getDelay(TimeUnit unit) {
+    return unit.convert(timeoutAt - System.nanoTime(), TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  public final int compareTo(Delayed that) {
+    long thisDelay = getDelay(TimeUnit.NANOSECONDS);
```

Comment:
Hmm... It'd be nice to use `timeoutAt` instead, to avoid calls to `System.nanoTime()`. It does cause some problem with `AsyncTimeout` being public, since it would be more convenient to assume `that instanceof AsyncTimeout`.